lib/ansible/modules/packaging/language/pip.py

```diff
@@ -235,6 +237,61 @@
                              'pip': 'import pkg_resources; print(pkg_resources.get_distribution("pip").version)'}
 
 
+class Package(Requirement):
+    def __init__(self, name_string, version_string=None):
+        if version_string is not None:
+            if version_string[0].isdigit():
+                name_string = "%s ==%s" % (name_string, version_string)
+            else:
+                name_string = "%s %s" % (name_string, version_string)
+        super(Package, self).__init__(name_string)
+
+    def have_version_specifier(self):
+        if len(self.specs) == 0:
+            return False
+        return True
+
+    def get_name(self):
+        return self.name.lower()
+
+    def satisfied_by(self, version_to_test):
+        return self.specifier.contains(version_to_test)
+
+    def install_command(self):
+        return str(self)
+
+
+def _is_valid_pkg_name(name):
+    try:
+        Requirement(name)
+        return True
+    except RequirementParseError:
+        return False
+
+
+def _fetch_one_pkg(name):
+    stack = list()
+    for ele in name:
+        if _is_valid_pkg_name(ele):
+            if len(stack) > 0:
+                yield ",".join(stack)
+            stack = [ele]
+        else:
+            stack.append(ele)
+    yield ",".join(stack)
+
+
+def _parses_name_input(module, name):
```

Comment:
> parses

`parse`, because it's an action. But I'd go for smth like `get_packages_from_specs`.

Reply:
I'd also turn `name` into `names` because it's a list and variable name should reflect this detail.